File: apps/mobile-app/src/db/models/Gasto.ts
import { Model } from '@nozbe/watermelondb';
import { field, text, date, readonly } from '@nozbe/watermelondb/decorators';

/**
 * Gasto del vendedor (combustible, peajes, comida, etc.).
 * Auto-aprobado al crearse. Foto del ticket opcional via attachments
 * (event_type='gasto', event_local_id=this.id).
 * Sync push hacia backend SyncRepository.UpsertGastoAsync.
 * comprobante_url es stampeado server-side post-upload del attachment.
 */
export default class Gasto extends Model {
  static table = 'gastos';

  @field('server_id') serverId!: number | null;
  @field('ruta_id') rutaId!: string | null;
  @field('ruta_server_id') rutaServerId!: number | null;
  @field('usuario_id') usuarioId!: number;
  @date('fecha_gasto') fechaGasto!: Date;
  @field('monto') monto!: number;
  @field('tipo_gasto') tipoGasto!: number; // TipoGasto enum (0=Combustible..99=Otro)
  @text('concepto') concepto!: string;
  @text('notas') notas!: string | null;
  @text('comprobante_url') comprobanteUrl!: string | null;
  @text('moneda') moneda!: string;
  @field('estado') estado!: number; // 0=Activo, 1=Invalidado
  @field('activo') activo!: boolean;
  @field('version') version!: number;
  @readonly @date('created_at') createdAt!: Date;
  @date('updated_at') updatedAt!: Date;
}

File: libs/HandySuites.Domain/Entities/Gasto.cs
using System.ComponentModel.DataAnnotations.Schema;
using HandySuites.Domain.Common;

namespace HandySuites.Domain.Entities;

public enum TipoGasto
{
    Combustible = 0,
    Peaje = 1,
    Comida = 2,
    Hospedaje = 3,
    Mantenimiento = 4,
    Estacionamiento = 5,
    Papeleria = 6,
    Otro = 99,
}

public enum EstadoGasto
{
    Activo = 0,
    Invalidado = 1,
}

/// <summary>
/// Registro de gasto incurrido por un vendedor durante su jornada/ruta.
/// Auto-aprobado al crearse: cuenta inmediatamente contra el corte de caja
/// de su ruta (descuenta de aRecibir en CierreRutaResumenDto). El supervisor
/// puede invalidar un gasto historicamente (no recomputa cierres ya cerrados).
/// Comprobante (foto del ticket) opcional, subido via attachments endpoint
/// y stampeado por MobileRecordId post-upload.
/// </summary>
[Table("Gastos")]
public class Gasto : AuditableEntity
{
    [Column("id")]
    public int Id { get; set; }

    [Column("tenant_id")]
    public int TenantId { get; set; }

    [Column("mobile_record_id")]
    public string? MobileRecordId { get; set; }

    /// <summary>Opcional: ruta a la que se imputa el gasto. Si null, vendedor lo registro fuera de ruta.</summary>
    [Column("ruta_id")]
    public int? RutaId { get; set; }

    [Column("usuario_id")]
    public int UsuarioId { get; set; }

    [Column("fecha_gasto")]
    public DateTime FechaGasto { get; set; } = DateTime.UtcNow;

    [Column("monto")]
    public decimal Monto { get; set; }

    [Column("tipo_gasto")]
    public TipoGasto TipoGasto { get; set; }

    [Column("concepto")]
    public string Concepto { get; set; } = string.Empty;

    [Column("notas")]
    public string? Notas { get; set; }

    /// <summary>URL del comprobante en Cloudinary o disco local. Stampeada por MobileAttachmentEndpoints tras upload.</summary>
    [Column("comprobante_url")]
    public string? ComprobanteUrl { get; set; }

    /// <summary>Moneda ISO-4217. Default 'MXN'. Futureproofing — multi-currency no soportado en v1.</summary>
    [Column("moneda")]
    public string Moneda { get; set; } = "MXN";

    [Column("estado")]
    public EstadoGasto Estado { get; set; } = EstadoGasto.Activo;

    [Column("invalidado_por")]
    public string? InvalidadoPor { get; set; }

    [Column("invalidado_en")]
    public DateTime? InvalidadoEn { get; set; }

    [Column("motivo_invalidacion")]
    public string? MotivoInvalidacion { get; set; }

    // Navigation properties
    public Tenant Tenant { get; set; } = null!;
    public RutaVendedor? Ruta { get; set; }
    public Usuario Usuario { get; set; } = null!;
}
